frontend/src/bundles/common/store/slice.ts

```diff
@@ -0,0 +1,50 @@
+import { createSlice } from '@reduxjs/toolkit';
+
+import {
+    type PayloadAction,
+    type ValueOf,
+} from '~/bundles/common/types/types.js';
+
+import { type Theme } from '../enums/theme.js';
+import { fetchTheme, setTheme } from './actions.js';
+
+type State = {
+    theme: ValueOf<typeof Theme> | null;
+    loading: boolean;
```

Comment:
why do you need loading state here? It's should completely synchronous

Reply:
Since we are using the storage object this returns a promise.
![image](https://github.com/BinaryStudioAcademy/bsa-winter-2023-2024-lime/assets/53876827/2dd777ce-35f6-488d-95ed-95d92e50573d)
Of course if I use the local storage directly it wouldn't need that